### PR TITLE
Add default configuration for individual builds and fix windows builds

### PR DESCRIPTION
**Pull request by johnoliver**

### CODE REVIEW

**Comment by karianna** [comment]
@johnoliver Groovy compile failed.

File: pipelines/build/common/create_job_from_template.groovy [hunk 1]
```diff
@@ -58,7 +58,25 @@ pipelineJob("$buildFolder/$JOB_NAME") {
...
+                <dt><b>SCM_REF</b></dt><dd>Source code ref to build, i.e branch, tag, commit id</dd>
+                <dt><b>BUILD_ARGS</b></dt><dd>args to pass to makejdk-any-platform.sh</dd>
+                <dt><b>NODE_LABEL</b></dt><dd>Labels of node to build on</dd>
+                <dt><b>CONFIGURE_ARGS</b></dt><dd>Arguments for ./configure</dd>
+                <dt><b>OVERRIDE_FILE_NAME_VERSION</b></dt><dd>Set the version string on the file name</dd>
+                <dt><b>RELEASE</b></dt><dd>Is this build a release</dd>
+                <dt><b>PUBLISH_NAME</b></dt><dd>Set name of publish</dd>
+                <dt><b>ADOPT_BUILD_NUMBER</b></dt><dd>Adopt build number</dd>
+                <dt><b>ENABLE_TESTS</b></dt><dd>Run tests</dd>
+                <dt><b>CLEAN_WORKSPACE</b></dt><dd>Wipe out workspace before build</dd>
```
Comment (karianna) [inline pipelines/build/common/create_job_from_template.groovy]: should use strong over b

File: sbin/build.sh [hunk 2]
```diff
@@ -397,7 +397,7 @@ buildSharedLibs() {
     local gradleJavaHome=$(getGradleHome)
     echo "Running gradle with $gradleJavaHome"
 
-    JAVA_HOME="$gradleJavaHome" GRADLE_USER_HOME=./gradle-cache ./gradlew --no-daemon clean uberjar
+    JAVA_HOME="$gradleJavaHome" GRADLE_USER_HOME=./gradle-cache bash ./gradlew --no-daemon clean uberjar
```
Comment (karianna) [inline sbin/build.sh]: Is bash on all platforms?

File: sbin/build.sh [hunk 3]
```diff
@@ -638,6 +636,12 @@ copyFreeFontForMacOS() {
   makeACopyOfLibFreeFontForMacOSX "${jreTargetPath}" "${BUILD_CONFIG[COPY_MACOSX_FREE_FONT_LIB_FOR_JRE_FLAG]}"
 }
 
+createTargetDirs() {
+  ## clean out old builds
+  rm -r "${BUILD_CONFIG[WORKSPACE_DIR]:?}/${BUILD_CONFIG[TARGET_DIR]}" || true
```
Comment (karianna) [inline sbin/build.sh]: I'd prefer this to be split into two seperate functions.

File: sbin/build.sh (hunk 2 above)
Comment (sxa555) [reply sbin/build.sh]: Much as I'd rather they weren't, our build scripts are full of fairly sophisticated bash-isms, so it has to be on all our systems already.

Comment (karianna) [reply sbin/build.sh]: Ack

**Pull request merged**